### PR TITLE
Cisco Stealthwatch - New Pack

### PR DESCRIPTION
**Pull request by ShiratG**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
A few sentences describing the overall goals of the pull request's commits.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Demisto
- [x] 5.0.0
- [ ] 5.5.0
- [ ] 6.0.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: Packs/CiscoStealthwatch/Integrations/CiscoStealthwatch/cisco_stealthwatch.py [hunk 1]
```diff
@@ -0,0 +1,613 @@
...
+        "endDateTime": end_time,
+        "recordLimit": limit,
+        "subject": {
+            "ipAddresses": {
+                "includes": ip_addresses if isinstance(ip_addresses, list) else [ip_addresses]
+            }
+        }
+    })
+    response = client.initialize_flow_search(tenant_id, data)
+    outputs = dict_safe_get(response, ['data', 'query'])
```
Comment (reutshal) [inline Packs/CiscoStealthwatch/Integrations/CiscoStealthwatch/cisco_stealthwatch.py:160]: Nice!

File: Packs/CiscoStealthwatch/Integrations/CiscoStealthwatch/cisco_stealthwatch.py [hunk 2]
```diff
@@ -0,0 +1,613 @@
...
+    table = tableToMarkdown('Query Flows Initializing Information:', outputs,
+                            headers=['id', 'status', 'percentComplete'], removeNull=True)
```
Comment (reutshal) [inline Packs/CiscoStealthwatch/Integrations/CiscoStealthwatch/cisco_stealthwatch.py:162]:
```suggestion
                            headers=['id', 'status', 'percentComplete'], removeNull=True, headerTransform=pascalToSpace)
```
better using this CSP transformer for a better headers view - 
image
True for all commands

File: Packs/CiscoStealthwatch/Integrations/CiscoStealthwatch/cisco_stealthwatch.py [hunk 3]
```diff
@@ -0,0 +1,613 @@
...
+    outputs = []
+    for data in dict_safe_get(response, ['data', 'flows']):
+        outputs.append(data)
+    headers = ['id', 'tenantId', 'flowCollectorId', 'protocol', 'serviceId', 'statistics', 'peer',
+               'subject']
+    table = tableToMarkdown('Query Flows Results Information:', outputs, headers=headers,
+                            removeNull=True)
+    return CommandResults(
+        outputs_prefix='CiscoStealthwatch.FlowResults',
+        outputs_key_field=['id', 'tenantId'],
```
Comment (reutshal) [inline Packs/CiscoStealthwatch/Integrations/CiscoStealthwatch/cisco_stealthwatch.py:218]: The id alone is not enough? did you check for duplicates?

File: Packs/CiscoStealthwatch/Integrations/CiscoStealthwatch/cisco_stealthwatch.py [hunk 4]
```diff
@@ -0,0 +1,613 @@
...
+        CommandResults: Raw response, outputs and readable outputs
+    """
+    response = client.list_tags(tenant_id)
+    outputs = []
+    for tag in response.get('data', []):
+        outputs.append(tag)
+
+    outputs = sorted(outputs, key=lambda x: x.get('id'))
+
+    table = tableToMarkdown(f'Tags (for tenant_id: {tenant_id}):', outputs,
```
Comment (reutshal) [inline Packs/CiscoStealthwatch/Integrations/CiscoStealthwatch/cisco_stealthwatch.py:242]:
```suggestion
    table = tableToMarkdown(f'Tags for tenant_id: {tenant_id}:', outputs,
```
No need for the brackets :)

File: Packs/CiscoStealthwatch/Integrations/CiscoStealthwatch/cisco_stealthwatch.py [hunk 5]
```diff
@@ -0,0 +1,613 @@
...
+        tenant_id (str): The id of the tenant to get its tag information
+        tag_id (str): The id of the tag to retrieve its information
+
+    Returns:
+        CommandResults: Raw response, outputs and readable outputs
+    """
+    response = client.get_tag(tenant_id, tag_id)
+    outputs = response.get('data', {})
+
+    table = tableToMarkdown(f'Tag (tenant_id: {tenant_id}, tag_id: {tag_id}):', outputs,
```
Comment (reutshal) [inline Packs/CiscoStealthwatch/Integrations/CiscoStealthwatch/cisco_stealthwatch.py:269]: Same here, I personally think that brackets don't look very good in a title, maybe something like this? 
`f'Tag {tag_id} with tenant id: {tenant_id} results:'`

Please fix in all commands human readable 🙏

File: Packs/CiscoStealthwatch/Integrations/CiscoStealthwatch/cisco_stealthwatch.py [hunk 6]
```diff
@@ -0,0 +1,613 @@
...
+        tenant_id (str): The id of the tenant to retrieve its information
+
+    Returns:
+        CommandResults: Raw response, outputs and readable outputs
+    """
+    if tenant_id:
+        response = client.get_tenant(tenant_id)
+        outputs = response.get('data', [])
+
+        table = tableToMarkdown(f'Tenant (tenant_id: {tenant_id})', outputs,
```
Comment (reutshal) [inline Packs/CiscoStealthwatch/Integrations/CiscoStealthwatch/cisco_stealthwatch.py:297]:
```suggestion
        table = tableToMarkdown(f'Results for tenant {tenant_id}:', outputs,
```

File: Packs/CiscoStealthwatch/Integrations/CiscoStealthwatch/cisco_stealthwatch.py [hunk 7]
```diff
@@ -0,0 +1,613 @@
...
+        tenant_id (str): The id of the tenant to retrieve its information
+        tag_id (str): The id of the tag to retrieve its information
+
+    Returns:
+        CommandResults: Raw response, outputs and readable outputs
+    """
+    response = client.tag_hourly_traffic(tenant_id, tag_id)
+    outputs = []
+    if response.get('data'):
+        for report in response['data'].get('data', []):
```
Comment (reutshal) [inline Packs/CiscoStealthwatch/Integrations/CiscoStealthwatch/cisco_stealthwatch.py:340]:
```suggestion
        for report in response.get('data', {}).get('data', []):
```

File: Packs/CiscoStealthwatch/Integrations/CiscoStealthwatch/cisco_stealthwatch.py [hunk 8]
```diff
@@ -0,0 +1,613 @@
...
+            report['tag_id'] = tag_id
+            report['tenant_id'] = tenant_id
+            value = report.get('value')
+            report.pop('value')
+            report.update(value)
```
Comment (reutshal) [inline Packs/CiscoStealthwatch/Integrations/CiscoStealthwatch/cisco_stealthwatch.py:345]: Why are we doing that? Please add a comment if still relevant

File: Packs/CiscoStealthwatch/Integrations/CiscoStealthwatch/cisco_stealthwatch.py [hunk 9]
```diff
@@ -0,0 +1,613 @@
...
+            outputs.append(report)
+
+    headers = ['timestamp', 'inboundByteCount', 'outboundByteCount', 'withinByteCount']
+    title = f'Hourly Tag Traffic Report (tenant_id: {tenant_id}, tag_id: {tag_id}):'
+    table = tableToMarkdown(title, outputs, headers=headers, removeNull=True)
+
+    return CommandResults(
+        outputs_prefix='CiscoStealthwatch.TagHourlyTraffic',
+        outputs_key_field=['tag_id', 'tenant_id', 'timestamp'],
```
Comment (reutshal) [inline Packs/CiscoStealthwatch/Integrations/CiscoStealthwatch/cisco_stealthwatch.py:354]: Do those three keys create the unique key together?

File: Packs/CiscoStealthwatch/Integrations/CiscoStealthwatch/cisco_stealthwatch.py [hunk 10]
```diff
@@ -0,0 +1,613 @@
...
+    Args:
+        client (Client): Cisco Stealthwatch Client
+        tenant_id (str): The id of the tenant of the security events process
+        search_id (str): The if of the search.
+
+    Returns:
+        CommandResults: Raw response, outputs and readable outputs
+    """
+    response = client.check_security_events_search_progress(tenant_id, search_id)
+    outputs = response.get('data')
```
Comment (reutshal) [inline Packs/CiscoStealthwatch/Integrations/CiscoStealthwatch/cisco_stealthwatch.py:450]:
```suggestion
    outputs = response.get('data', {})
```

File: Packs/CiscoStealthwatch/Integrations/CiscoStealthwatch/cisco_stealthwatch.py [hunk 11]
```diff
@@ -0,0 +1,613 @@
...
+    # get the service API url
+    base_url = params.get('server_url')
+
+    verify_certificate = not params.get('insecure', False)
+
+    proxy = params.get('proxy', False)
+
+    command = demisto.command()
+
+    LOG(f'Command being called is {command}')
```
Comment (reutshal) [inline Packs/CiscoStealthwatch/Integrations/CiscoStealthwatch/cisco_stealthwatch.py:552]:
```suggestion
    demisto.info(f'Command being called is {command}')
```

File: Packs/CiscoStealthwatch/Integrations/CiscoStealthwatch/cisco_stealthwatch_description.md [hunk 12]
```diff
@@ -0,0 +1,14 @@
+## Configure Cisco Stealthwatch on Cortex XSOAR
```
Comment (reutshal) [inline Packs/CiscoStealthwatch/Integrations/CiscoStealthwatch/cisco_stealthwatch_description.md:1]: This looks more like a readme file, you can read about the integration_description file here: https://xsoar.pan.dev/docs/documentation/integration-description
Of course @ShirleyDenkberg would know best

File: Packs/CiscoStealthwatch/Integrations/CiscoStealthwatch/cisco_stealthwatch_test.py [hunk 13]
```diff
@@ -0,0 +1,189 @@
...
+                      json=mock_response)
+    client = Client(base_url='https://', auth=('', ''), verify=False, proxy=False)
+    response = cisco_stealthwatch_get_top_alarming_tags_command(client, 'x')
+
+    assert response.outputs_prefix == 'CiscoStealthwatch.AlarmingTag'
+    assert response.outputs_key_field == ['tenant_id', 'hostGroupIds']
+    assert response.outputs[0].get('tenant_id') == 'x'
+
+
+# # def test_cisco_stealthwatch_list_alarms_command(requests_mock):
```
Comment (reutshal) [inline Packs/CiscoStealthwatch/Integrations/CiscoStealthwatch/cisco_stealthwatch_test.py:140]: Please delete if not relevant

File: Packs/CiscoStealthwatch/Integrations/CiscoStealthwatch/test_data/list_security_events_results.json [hunk 14]
```diff
@@ -0,0 +1 @@
...dress":"172.16.100.4","port":0,"protocol":"tcp","tags":[{"name":"Catch All","id":65534}]},"target":{"ipAddress":"192.168.91.12","port":445,"protocol":"tcp","tags":[{"name":"Catch All","id":65534}]},"details":[{"key":"source_host@username","value":""},{"key":"source_host@policy_id","value":"1"},{"key":"source_host@mac_address","value":""},{"key":"target_host@username","value":""},{"key":"target_host@policy_id","value":"1"},{"key":"target_host@mac_address","value":""},{"key":"category_points@high-concern-index","value":"248994"},{"key":"category_points@high-target-index","value":"248994"},{"key":"category_points@high-recon-index","value":"248994"},{"key":"baseline@baseline","value":"0"},{"key":"baseline@threshold","value":"0"},{"key":"baseline@current_value","value":"0"},{"key":"baseline@tolerance","value":"0"},{"key":"flow@protocol","value":"6"},{"key":"flow@service","value":"29"},{"key":"flow@source_port","value":"0"},{"key":"flow@target_port","value":"445"},{"key":"flow@event_port","value":"445"},{"key":"flow@flow_id","value":"0"},{"key":"flow@source_is_server","value":"false"},{"key":"targetIPAddress","value":"192.168.91.12"},{"key":"points","value":"248994"}],"hitCount":1537}]}}
```
Comment (reutshal) [inline Packs/CiscoStealthwatch/Integrations/CiscoStealthwatch/test_data/list_security_events_results.json:1]: Are you sure there are no secrets here?

File: Packs/CiscoStealthwatch/Playbooks/playbook-List_Cisco_Stealthwatch_Security_Events_README.md [hunk 15]
```diff
@@ -0,0 +1,39 @@
...
+
+## Playbook Outputs
+---
+
+| **Path** | **Description** | **Type** |
+| --- | --- | --- |
+| CiscoStealthwatch.SecurityEventResults | The result of the search | unknown |
+
+## Playbook Image
+---
```
Comment (reutshal) [inline Packs/CiscoStealthwatch/Playbooks/playbook-List_Cisco_Stealthwatch_Security_Events_README.md:39]: Please provide a playbook image. You can do it like described here:
https://xsoar.pan.dev/docs/documentation/readme_file#images

File: Packs/CiscoStealthwatch/Playbooks/playbook-Query_Cisco_Stealthwatch_Flows_README.md [hunk 16]
```diff
@@ -0,0 +1,40 @@
...
+## Playbook Outputs
+---
+
+| **Path** | **Description** | **Type** |
+| --- | --- | --- |
+| CiscoStealthwatch.FlowResults | The result of the search | unknown |
+
+## Playbook Image
+---
+![Query Cisco Stealthwatch Flows](Insert the link to your image here)
```
Comment (reutshal) [inline Packs/CiscoStealthwatch/Playbooks/playbook-Query_Cisco_Stealthwatch_Flows_README.md:40]: Please insert an image

File: Packs/CiscoStealthwatch/Integrations/CiscoStealthwatch/cisco_stealthwatch.yml [hunk 17]
```diff
@@ -0,0 +1,410 @@
...
+  dockerimage: demisto/python3:3.9.1.15759
+  feed: false
+  isfetch: false
+  longRunning: false
+  longRunningPort: false
+  runonce: false
+  script: '-'
+  subtype: python3
+  type: python
+fromversion: 5.5.0
```
Comment (reutshal) [inline Packs/CiscoStealthwatch/Integrations/CiscoStealthwatch/cisco_stealthwatch.yml:410]: Please add the test playbook here like that: 

```suggestion
tests:
  - Yout test playbook ID
fromversion: 5.5.0
```

File: Packs/CiscoStealthwatch/TestPlaybooks/playbook-Cisco_Stealthwatch-Test.yml [hunk 18]
```diff
@@ -0,0 +1,992 @@
+id: Cisco Stealthwatch
```
Comment (reutshal) [inline Packs/CiscoStealthwatch/TestPlaybooks/playbook-Cisco_Stealthwatch-Test.yml:1]:
```suggestion
id: Cisco Stealthwatch Test
```

File: Packs/CiscoStealthwatch/TestPlaybooks/playbook-Cisco_Stealthwatch-Test.yml [hunk 19]
```diff
@@ -0,0 +1,992 @@
...
+version: -1
+vcShouldKeepItemLegacyProdMachine: false
+name: Cisco Stealthwatch
```
Comment (reutshal) [inline Packs/CiscoStealthwatch/TestPlaybooks/playbook-Cisco_Stealthwatch-Test.yml:4]:
```suggestion
name: Cisco Stealthwatch Test
```

**Review by reutshal — CHANGES_REQUESTED** [review]
Awesome coding!
What about classifiers+ mappers + layouts?

**Comment by reutshal** [comment]
@kirbles19 @ShirleyDenkberg Please help with the docs review 🙏

File: Packs/CiscoStealthwatch/Integrations/CiscoStealthwatch/cisco_stealthwatch.py (hunk 2 above)
Comment (ShiratG) [reply Packs/CiscoStealthwatch/Integrations/CiscoStealthwatch/cisco_stealthwatch.py:162]: Understood, but what is CSP?

File: Packs/CiscoStealthwatch/Integrations/CiscoStealthwatch/cisco_stealthwatch.py (hunk 3 above)
Comment (ShiratG) [reply Packs/CiscoStealthwatch/Integrations/CiscoStealthwatch/cisco_stealthwatch.py:218]: I re-checked it and the `id` here is enough, but on other commands, like `cisco_stealthwatch_get_top_alarming_tags_command` and `cisco_stealthwatch_get_tag_hourly_traffic_report_command`, there is no id to rely on, so I used a combination of the `tag id`, `tenant id`, and the `timestamp` to make sure there is no missing information

File: Packs/CiscoStealthwatch/Integrations/CiscoStealthwatch/cisco_stealthwatch.py (hunk 2 above)
Comment (reutshal) [reply Packs/CiscoStealthwatch/Integrations/CiscoStealthwatch/cisco_stealthwatch.py]: CommonServerPython :) 
That's where we have all of the methods that are commonly used in our scripts + integration, it is imported in every integration.

File: Packs/CiscoStealthwatch/Integrations/CiscoStealthwatch/test_data/list_security_events_results.json (hunk 14 above)
Comment (ShiratG) [reply Packs/CiscoStealthwatch/Integrations/CiscoStealthwatch/test_data/list_security_events_results.json:1]: I removed IP addresses and the company name just to be sure, thanks!

**Review by reutshal — APPROVED** [review]

**Comment by ShirleyDenkberg** [comment]
@reutshal Doc review is complete.  I sent you the following updated files by Slack since I do not have permission to edit the files in this PR. 
- Cisco Stealthwatch (integration yml)
- playbook-List_Cisco_Stealthwatch_Security_Events
- playbook-Query_Cisco_Stealthwatch_Flows
The README files for the integration and playbooks should be regenerated since I made updates to the yml files.
In addition, I sent you the pack README which is new.

**Review by reutshal — APPROVED** [review]

**Pull request merged**